src/jquery.multiselect.js

```diff
@@ -579,71 +623,116 @@
-          var lastChar = minVal[ minVal.length -1 ];
-          width = minVal.match(/\d+/);
-          if(lastChar === '%') {
-            width = this.element.parent().outerWidth() * (width/100);      // element is a jQuery object
-          } else {
-            width = parseInt(minVal, 10);
-          }
-          break;
+         if (minWidth === 'auto') {
+            $menu = self.$menu;
+            var appendEl = self._getAppendEl();
+            var cssVisibility = $menu.css('visibility');
+            var cssWidth = $menu.css('width');
+            var cssDisplay = $menu.css('display');
+            var cssulDisplay = $menu.find('ul:eq(0)').css('display');
+            $menu.appendTo('body').css({visibility:'hidden', width:'auto', display:'inline'}).find('ul').css('display', 'inline');
+            var autoWidth = $menu.width();
+            $menu.css({visibility:cssVisibility, width:cssWidth, display:cssDisplay}).find('ul').css('display', cssulDisplay).appendTo( appendEl );
+            console.log('AUTO WIDTH: ', cssWidth, autoWidth);
+            return autoWidth;
+         }
+         width = parseInt(minWidth, 10);
+
+         if ( minWidth.slice(-1) === '%' )
+            width = self.element.parent().innerWidth() * (width/100);     // element is a jQuery object.
       }
+
       return width;
     },
 
-    // set button width
-    _setButtonWidth: function() {
-      var width = this.element.outerWidth();    // element is a jQuery object
-      var minVal = this._getMinWidth();
+    // set button width.   jQuery seems to be required here, as getting the outerWidth by the non-jQuery way is unreliable.
+    _setButtonWidth: function(recalc) {
+      var self = this;
+      if (self._savedButtonWidth && !recalc)
+         return;
+
+      self._positioned = false;               // Force menu positioning to be adjusted on next open.
+      var width = self.element.outerWidth();  // element is a jQuery object
+      var minWidth = self._getMinWidth();
 
-      if(width < minVal) {
-        width = minVal;
-      }
       // set widths
-      this.$button.outerWidth(width);
+      self._savedButtonWidth = width < minWidth ? minWidth : width;   // The button width is cached.
+      self.$button.outerWidth(self._savedButtonWidth);
     },
 
     // set menu width
-    _setMenuWidth: function() {
-      var width = (this.$button.outerWidth() <= 0) ? this._getMinWidth() : this.$button.outerWidth();
-      this.$menu.outerWidth(this.options.menuWidth || width);
+    _setMenuWidth: function(recalc) {
+      var self = this;
+      if (self._savedMenuWidth && !recalc)
+         return;
+
+      self._positioned = false;             // Force menu positioning to be adjusted on next open.
+      var width = self.options.menuWidth;
+
+      if (!width) {                                                   // Exact width not provided; determine appropriate width.
+         width = self._savedButtonWidth || self.$button.outerWidth();
```

Comment:
You're in the recalc section here, and still using the saved value. Is that intentional?

Reply:
It is referencing a button width in the setMenuWidth function.  Since the button width was just likely already calculated, this just tries to use the recently recalculated cached value.

Reply:
Ah, I see. That makes sense.